atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -83,18 +83,49 @@ def run(self):
             if nvr:
                 self._base_image_build = self.wait_for_parent_image_build(nvr)
 
+        manifest_list_mismatches = []
         for img, local_tag in self.workflow.builder.parent_images.items():
             if base_image_is_custom(img.to_str()):
                 continue
 
             nvr = self.detect_parent_image_nvr(local_tag) if local_tag else None
             if nvr == self._base_image_nvr:  # don't look up base image a second time
                 self._parent_builds[img] = self._base_image_build
-                continue
-            self._parent_builds[img] = self.wait_for_parent_image_build(nvr) if nvr else None
-
+            else:
+                self._parent_builds[img] = self.wait_for_parent_image_build(nvr) if nvr else None
+
+            if self._parent_builds[img]:
+                mismatch = self.check_manifest_list_digest(local_tag, self._parent_builds[img])
+                if mismatch:
+                    manifest_list_mismatches.append(mismatch)
+            else:
```

Comment:
this can happen only when base image (last parent image) is FROM scratch or custom_base_image

IF build can't be find in koji, wait_for_parent_image_build will raise error

so I'd just remove that else section with error message (or write there that base image is scratch or custom, and we don't expect that exist in koji)

Reply:
How are parent images from other stages checked against existence of koji build? IIUC, only base_image is checked for  being in koji (i.e. last FROM statement). How about other stages? IIRC then parents from other stages may pass even if they aren't in koji.

Reply:
1st it checks for base image (if not scratch or custom) with wait_for_parent_image_build line 84

then it checks for each other parent with wait_for_parent_image_build line 95

Reply:
* line 91: `detect_parent_image_nvr` returns `None` when NVR is not found, because of this:
```
        if len(label_values) != len(label_names):  # don't have all the necessary labels
            self.log.info("Image '%s' NVR missing; not searching for Koji build.", image_name)
            return None
``` 
* line 95: `self._parent_builds[img] = self.wait_for_parent_image_build(nvr) if nvr else None`
`nvr is None`, no `wait_for_parent_image_build` is executed, `None` is assigned to `_parent_builds[img]`

* line 97: mismatch check is not executed because of `None`

Reply:
Oh, but the same applies for `_base_image_build`

Reply:
yeah true, but osbs build will fail when any of those labels are missing, so you can't have in koji build which is missing them :)

Reply:
OMG, that is just relying on side effects from other plugins

Reply:
mmm actually what I am thinking is, that we should report IF nvr is None,
and also consider it as not found in koji (as that is some weird image, not build by osbs)

cc @athos-ribeiro

Reply:
AFAIK, if `we don't allow non-koji images` this is the best place for killing that build when NVR is not found.

Reply:
> mmm actually what I am thinking is, that we should report IF nvr is None,
> and also consider it as not found in koji (as that is some weird image, not build by osbs)
> 
> cc @athos-ribeiro

If the NVR is not found, we are missing proper Labels. When will the build fail? This does pass in osbs-box when using a Fedora image without the propar labels (and as consequence, we cannot check for digests).

Reply:
I meant that build of such parent would fail with osbs

Reply:
Ok, so we should raise RuntimeError if nvr is None and the build is neither custom base nor from scrach. is that it?

@rcerven @MartinBasti.

Warning: this will break osbs-box, but seems to be the safe approach

Reply:
This is done now. @cverna would this break Fedora workflow?

Reply:
Yes this will most likely break osbs in Fedora, since we are not building the Fedora base image in OSBS for the following reasons
 - lack of OSBS cluster for non x86_64 arches
 - cannot publish images to Docker Hub official library using OSBS.

We are currently working on getting an aarch64 cluster but we are low on hardware resources, and I think it will take a while for us to have a ppc64le, armv7 and s390x cluster.

Reply:
Thanks for the feedback, @cverna.

I will follow-up with a PR adding an option to enable builds with external (non-osbs-built) base images.